### ISSUE TITLE
`TypeApplier` fails to annotate consecutive identically named variables in separate functions

### ISSUE THREAD

**Issue by bengsparks** [saltudelft/libsa4py#35]
MRE: 

```py
import textwrap
from pprint import pprint

import libcst
from libcst import metadata

from libsa4py.cst_extractor import Extractor
from libsa4py.cst_transformers import TypeAnnotationRemover, TypeApplier


mod = libcst.parse_module(
    textwrap.dedent(
        """
def f():
    x: int = 10

def g():
    x: str = "Hello World"
"""
    )
)
mod_annotations = Extractor.extract(mod.code).to_dict()
pprint(mod_annotations["funcs"])

annoless = mod.visit(TypeAnnotationRemover())
annotated = metadata.MetadataWrapper(annoless, unsafe_skip_copy=True).visit(
    TypeApplier(mod_annotations, apply_nlp=False)
)
new_annotations = Extractor.extract(annotated.code).to_dict()
pprint(new_annotations["funcs"])

```

Output: 
```py 
[{'docstring': {'func': None, 'long_descr': None, 'ret': None},
  'fn_lc': ((2, 0), (3, 24)),
  'fn_var_ln': {'x': ((3, 4), (3, 5))},
  'fn_var_occur': {'x': [['x', 'builtins', 'int']]},
  'name': 'f',
  'params': {},
  'params_descr': {},
  'params_occur': {},
  'q_name': 'f',
  'ret_exprs': [],
  'ret_type': '',
  'variables': {'x': 'builtins.int'}},
 {'docstring': {'func': None, 'long_descr': None, 'ret': None},
  'fn_lc': ((5, 0), (6, 35)),
  'fn_var_ln': {'x': ((6, 4), (6, 5))},
  'fn_var_occur': {'x': [['x', 'builtins', 'str']]},
  'name': 'g',
  'params': {},
  'params_descr': {},
  'params_occur': {},
  'q_name': 'g',
  'ret_exprs': [],
  'ret_type': '',
  'variables': {'x': 'builtins.str'}}]
[{'docstring': {'func': None, 'long_descr': None, 'ret': None},
  'fn_lc': ((3, 0), (4, 24)),
  'fn_var_ln': {'x': ((4, 4), (4, 5))},
  'fn_var_occur': {'x': [['x', 'builtins', 'int']]},
  'name': 'f',
  'params': {},
  'params_descr': {},
  'params_occur': {},
  'q_name': 'f',
  'ret_exprs': [],
  'ret_type': '',
  'variables': {'x': 'builtins.int'}},
 {'docstring': {'func': None, 'long_descr': None, 'ret': None},
  'fn_lc': ((6, 0), (7, 21)),
  'fn_var_ln': {'x': ((7, 4), (7, 5))},
  'fn_var_occur': {'x': []},
  'name': 'g',
  'params': {},
  'params_descr': {},
  'params_occur': {},
  'q_name': 'g',
  'ret_exprs': [],
  'ret_type': '',
  'variables': {'x': ''}}]
  ```

Note that the second dict for the function `g` does not contain a type for the variable `x`, despite the original clearly finding the anntotation `builtins.str`

A tentative fix that I applied was to add the following to the `leave_FunctionDef` method of `TypeApplier`, but I am unsure of its correctness.
```py 
self.last_visited_assign_t_name = None
self.last_visited_assign_t_count = 0
```
Some resetting of `self.last_visited_assign_t_name` seems to be in order, or perhaps this should be assigned a fully qualified name instead of just the variable's name.

I hope this helps (or that I have made a mistake :) )

**Comment by mir-am** [comment]
Thanks for reporting an issue.
I will investigate it as soon as possible.

**Comment by mir-am** [comment]
@bengsparks
The fix is now included in the newest release, `v0.4.0`. Check it out!
